docs/third-party-developers/extensibility/rest-api/extend-rest-api-add-custom-fields.md

```diff
@@ -0,0 +1,176 @@
+
+/**
+ * Class for integrating with WooCommerce Blocks
+ */
+class Custom_Inner_Block_Blocks_Integration implements IntegrationInterface {
+	// ... Some code here
+
+	/**
+	 * When called invokes any initialization/setup for the integration.
+	 */
+	public function initialize() {
+		require_once __DIR__ . '/custom-inner-block-extend-store-endpoint.php';
+		// ... Some code here: (e.g. init functions that registers scripts and styles, and other instructions)
+		$this->extend_store_api();
+	}
+
+	/**
+	 * Extends the cart schema to include the custom-inner-block value.
+	 */
+	private function extend_store_api() {
+		Custom_Inner_Block_Extend_Store_Endpoint::init();
+	}
+}
+```
+
+- The `custom-inner-block-extend-store-endpoint.php` file: extends the [Store API](https://github.com/woocommerce/woocommerce-blocks/tree/trunk/src/StoreApi) and adds hooks to save and display your new field block instructions.
+
+```php
+use Automattic\WooCommerce\Blocks\Package;
+use Automattic\WooCommerce\Blocks\StoreApi\Schemas\CartSchema;
+use Automattic\WooCommerce\Blocks\StoreApi\Schemas\CheckoutSchema;
+
+/**
+ * Your New Field Block Extend Store API.
+ */
+class Custom_Inner_Block_Extend_Store_Endpoint {
+	/**
+	 * Stores Rest Extending instance.
+	 *
+	 * @var ExtendRestApi
+	 */
+	private static $extend;
+
+	/**
+	 * Plugin Identifier, unique to each plugin.
+	 *
+	 * @var string
+	 */
+	const IDENTIFIER = 'new-field-block';
+
+	/**
+	 * Bootstraps the class and hooks required data.
+	 *
+	 */
+	public static function init() {
+		self::$extend = Automattic\WooCommerce\StoreApi\StoreApi::container()->get( Automattic\WooCommerce\StoreApi\Schemas\ExtendSchema::class );
+		self::extend_store();
+	}
+
+	/**
+	 * Registers the actual data into each endpoint.
+	 */
+	public static function extend_store() {
+		if ( is_callable( [ self::$extend, 'register_endpoint_data' ] ) ) {
+			self::$extend->register_endpoint_data(
+				[
+					'endpoint'        => CheckoutSchema::IDENTIFIER,
+					'namespace'       => self::IDENTIFIER,
+					'schema_callback' => [ 'Custom_Inner_Block_Extend_Store_Endpoint', 'extend_checkout_schema' ],
+					'schema_type'     => ARRAY_A,
+				]
+			);
+		}
+	}
+
+	/**
+	 * Register shipping workshop schema into the Checkout endpoint.
+	 *
+	 * @return array Registered schema.
+	 *
+	 */
+	public static function extend_checkout_schema() {
+		return [
+            'Value_1'   => [
+                'description' => // ... description of the field
+                'type'        => // ... type of the field, this should be a string
+                'context'     => // ... context of the field, this should be an array containing 'view' and 'edit'
+                'readonly'    => // ... whether the field is readonly or not, this should be a boolean
+                'optional'    => // ... whether the field is optional or not, this should be a boolean
+            ],
+			// ... other values
+        ];
+	}
+}
+```
+
+## Conclusion
```

Comment:
Before we conclude, we should have an example of how to _get_ this custom field's data while processing the checkout, for example in the `woocommerce_store_api_checkout_update_order_from_request` hook.